src/parser/core/Combatant.ts

```diff
@@ -276,8 +276,12 @@ class Combatant extends Entity {
     return Boolean(this.conduitsByConduitID[spellId]);
   }
 
-  conduitRankBySpellID(spellId: number) {
-    return this.conduitsByConduitID[spellId] && this.conduitsByConduitID[spellId].rank;
+  conduitRankBySpellID(spellId: number): number {
+    if (!(spellId in this.conduitRankBySpellID)) {
+      return 0;
```

Comment:
this is now a slight behaviour change. previously it would return `undefined`:

```js
const conduitsByConduitID = {
    1: {
        rank: 2
    }
}

conduitsByConduitID[2] && conduitsByConduitID[2].rank // undefined
conduitsByConduitID[1] && conduitsByConduitID[1].rank // 2
```

this error was however uncaught, ts happily thought its a `number` either way...

Reply:
would honestly prefer this to return undefined, but i know we've had errors pop up related to this and that TS didn't catch it means that I'd be spooked about switching the return type.

which is to say, this change seems okay to me (especially since this goes away entirely for 10.0 :smiley:)